functions/minsurf_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package functions
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/diff/fd"
+	"github.com/gonum/floats"
+)
+
+func TestMinimalSurface(t *testing.T) {
```

Comment:
For two reasons
1. The only alternative implementation I have is the 20+ years old MINPACK code. I did not feel like fiddling with it to make it compile, run and extract a reference solution.
2. Even if I had a reference X, I did not want to dump those large vectors into the code. I admit that this is not a serious reason.